tezos/interop/src/ffi.rs

```diff
@@ -65,38 +65,39 @@ mod tezos_ffi {
 }
 
 /// Initializes the ocaml runtime and the tezos-ffi callback mechanism.
-pub fn setup() {
+pub fn setup() -> OCamlRuntime {
     static INIT: Once = Once::new();
+    let ocaml_runtime = OCamlRuntime::init();
 
     INIT.call_once(|| {
-        ocaml_interop::OCamlRuntime::init_persistent();
         tezos_interop_callback::initialize_callbacks();
     });
+
+    ocaml_runtime
 }
 
 /// Tries to shutdown ocaml runtime gracefully - give chance to close resources, trigger GC finalization...
 ///
 /// https://caml.inria.fr/pub/docs/manual-ocaml/intfc.html#sec467
 pub fn shutdown() {
-    ocaml_interop::OCamlRuntime::shutdown_persistent()
+    // TODO
```

Comment:
Note, this is done now when the executor thread finishes, but I have to revise all these parts and think more about the management the new runtime handle.